src/hermite/cubic.jl

```diff
@@ -0,0 +1,163 @@
+struct CubicHermite
+    xs::Vector{Float64}
+    ys::Vector{Float64}
+    dydxs::Vector{Float64}
+    function CubicHermite(xs, ys, dydxs)
+        if length(xs) < 2
+            throw(
+                DomainError(
+                    "There must be at least two samples in the domain of the CubicHermite interpolator.",
+                ),
+            )
+        end
+        x = xs[1]
+        for i = 2:length(xs)
+            if xs[i] <= x
+                throw(DomainError("The list of abscissas must be strictly increasing."))
+            end
+            x = xs[i]
+        end
+        if (length(xs) != length(ys) || length(ys) != length(dydxs))
+            throw(
+                DomainError(
+                    "There must be exactly the same number of abscissasas ordinates and derivatives.",
+                ),
+            )
+        end
+        new(xs, ys, dydxs)
+    end
+end
+
+function (ch::CubicHermite)(x::Float64)::Float64
+    if x <= ch.xs[1]
+        if x == ch.xs[1]
+            return ch.ys[1]
+        end
+        throw(
+            DomainError(
+                "Requested abscissas $x is less than the minimum value of the domain of the interpolator.",
+            ),
```

Comment:
Test is needed here to make sure this error is thrown.

Reply:
Fixed.